src/Compilers/CSharp/Portable/Parser/DirectiveParser.cs

```diff
@@ -467,15 +467,10 @@ private DirectiveTriviaSyntax ParsePragmaDirective(SyntaxToken hash, SyntaxToken
             {
                 var warning = this.EatContextualToken(SyntaxKind.WarningKeyword);
                 SyntaxToken style;
-                if (this.CurrentToken.Kind == SyntaxKind.DisableKeyword || this.CurrentToken.Kind == SyntaxKind.RestoreKeyword ||
-                    this.CurrentToken.Kind == SyntaxKind.EnableKeyword)
```

Comment:
>EnableKeyword [](start = 57, length = 13)

I would probably have left the syntax, and just moved `IDS_FeaturePragmaWarningEnable` to be attached to LangVersion.Preview.